packages/diagrams/frontend/sirius-components-diagrams/src/renderer/panel/DiagramPanelContribution.tsx

```diff
@@ -0,0 +1,77 @@
+/*******************************************************************************
+ * Copyright (c) 2023, 2024 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+
+import ClickAwayListener from '@material-ui/core/ClickAwayListener';
+import IconButton from '@material-ui/core/IconButton';
+import Paper from '@material-ui/core/Paper';
+import Popper from '@material-ui/core/Popper';
+import Tooltip from '@material-ui/core/Tooltip';
+
+import { Selection, useSelection } from '@eclipse-sirius/sirius-components-core';
+import { useRef, useState } from 'react';
+import { DiagramPanelContributionProps } from './DiagramPanelContribution.types';
+
+export const DiagramPanelContribution = ({ view, editingContextId, diagramId }: DiagramPanelContributionProps) => {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+  const { setSelection } = useSelection();
+
+  const handlePanel = () => {
+    const newSelection: Selection = {
+      entries: [
+        {
+          id: diagramId,
+          label: '',
+          kind: '',
+        },
+      ],
+    };
+    setSelection(newSelection);
```

Comment:
This is a hack because `FormBasedView` uses the current selection to build the event input. We saw two solutions to address this:
- Make `objectIds` an optional parameter of `FormBasedView` to allow consumers to specify on which elements we want to open the subscription
- Select the diagram before we create the view so the current selection matches what the backend expects.

We went for the second one because we didn't want to change the API of `FormBasedView`, but we can go for another solution, including a different "FormBasedView" tailored to the DiagramFilter (see my other comment on why we use `PropertiesEventInput` as input of the subscription)